### PR TITLE
SRIOV VFs Naming

### PR DESCRIPTION
**Pull request by rleon**

### CODE REVIEW

File: kernel-boot/rdma_rename.c [hunk 1]
```diff
@@ -217,13 +234,135 @@ static int is_pci_ari_enabled(char *devname)
...
+	DIR *dir;
+	int ret;
+
+	/* Check if this is a virtual function. */
+	ret = asprintf(&physfn_pcidev, "%s/physfn", p->pcidev);
+	if (ret < 0)
+		return -ENOMEM;
+
+	ret = lstat(physfn_pcidev, &st);
+	if (ret) {
```
[the remark this answers is not in the document]
Comment (rleon) [reply kernel-boot/rdma_rename.c]: I used lstat to test the existence of symlink.

File: kernel-boot/rdma_rename.c [hunk 2]
```diff
@@ -217,13 +234,135 @@ static int is_pci_ari_enabled(char *devname)
...
+
+	while ((dent = readdir(dir))) {
+		const char *s = "virtfn";
+		struct pci_info v = {};
+
+		if (strncmp(dent->d_name, s, strlen(s)) ||
+		    strlen(dent->d_name) == strlen(s))
+			continue;
+
+		ret = asprintf(&v.pcidev, "%s/%s", physfn_pcidev, dent->d_name);
```
[the remark this answers is not in the document]
Comment (rleon) [reply kernel-boot/rdma_rename.c]: Yes, I think that it leaks,

Regarding readlinkat - no reason - simple copy/paste

**Comment by rleon** [comment]
* removed lstat
* added missing free(), didn't change to use readlinkat, because didn't want to change fill_pcin_info function.
* converted all PIC identifier to use unsigned int.

**Comment by jgunthorpe** [comment]
The checkpatch warnings about unsigned should be fixed

**Comment by rleon** [comment]
> The checkpatch warnings about unsigned should be fixed

Why didn't checks fail? I relied on Travis and Azure to catch it.

**Comment by rleon** [comment]
* Fixed checkpatch warnings

**Comment by jgunthorpe** [comment]
> > The checkpatch warnings about unsigned should be fixed
> 
> Why didn't checks fail? I relied on Travis and Azure to catch it.

Azure caught it, but the top level checks screen shows a green check anyhow. The 2nd level screen shows there were warnings. I decided to make the sparse stuff warnings as we so often see them as false-positives.

**Pull request merged**